pennylane/qnode.py

```diff
@@ -582,9 +605,15 @@ def jacobian(self, params, which=None, *, method='B', h=1e-7, order=1, **kwargs)
         if isinstance(params, numbers.Number):
             params = (params,)
 
-        if not self.ops:
+        # remove jacobian specific keyword arguments
+        circuit_kwargs = {}
+        circuit_kwargs.update(kwargs)
+        for k in ('h', 'order', 'shots', 'force_order2'):
+            circuit_kwargs.pop(k, None)
```

Comment:
I think there should be a better way of separating the kwargs meant for `QNode.func`, and the ones meant for QNode methods. Maybe `jacobian` should receive the `QNode.func` kwargs as a separate dictionary argument?

Reply:
I don't think this is possible, purely because autograd will always send the function keyword arguments as keyword arguments. What would work is instead to collect the Jacobian keyword arguments within a `grad_options` dictionary.

Reply:
I thought about changing this here, but it seems out of scope. Should we mark this for a new PR?